torch/csrc/utils/python_numbers.h

```diff
@@ -109,13 +109,12 @@ inline double THPUtils_unpackDouble(PyObject* obj) {
   if (PyLong_Check(obj)) {
     int overflow;
     long long value = PyLong_AsLongLongAndOverflow(obj, &overflow);
```

Comment:
I'm not sure this part of the function is needed anymore. What would happen if we immediately converted using PyFloat_AsDouble?

Reply:
I tested removing the `PyLong_Check` part of code and immediately converting using `PyFloat_AsDouble` locally, it has the same result as casting the result of `PyLong_AsLongLongAndOverflow` to `double` when overflow and precision lost checks pass.